platform/solr/solr-password/src/main/java/ddf/platform/solr/security/SolrPasswordUpdateImpl.java

```diff
@@ -0,0 +1,221 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.platform.solr.security;
+
+import static javax.ws.rs.core.Response.Status.Family.SUCCESSFUL;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.security.common.audit.SecurityLogger;
+import ddf.security.encryption.EncryptionService;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import javax.annotation.Nullable;
+import javax.ws.rs.core.Response.StatusType;
+import org.apache.commons.lang.StringUtils;
+import org.apache.felix.utils.properties.Properties;
+import org.codice.ddf.cxf.client.ClientFactoryFactory;
+import org.codice.ddf.cxf.client.SecureCxfClientFactory;
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolrPasswordUpdateImpl implements SolrPasswordUpdate {
+
+  private static final String SYSTEM_PROPERTIES_FILE = "custom.system.properties";
+  private static final String KARAF_ETC = "karaf.etc";
+  private static final String SET_USER_JSON_TEMPLATE = "{ \"set-user\": {\"%s\" : \"%s\"}}";
+  private static final Logger LOGGER = LoggerFactory.getLogger(SolrPasswordUpdateImpl.class);
+
+  @SuppressWarnings("squid:S2068" /* This constant does hold an actual password */)
+  private static final String SOLR_PASSWORD_PROPERTY_NAME = "solr.password";
+
+  private final ClientFactoryFactory restClientFactoryFactory;
+  private final UuidGenerator uuidGenerator;
+  private final EncryptionService encryptionService;
+  private SolrAuthResource solrAuthResource;
+  private String newPasswordPlainText;
+  private StatusType solrResponse = null;
+  private String newPasswordWrappedEncrypted;
+  private boolean passwordSavedSuccessfully = false;
+
+  public SolrPasswordUpdateImpl(
+      UuidGenerator uuidGenerator,
+      ClientFactoryFactory clientFactoryFactory,
+      EncryptionService encryptionService) {
+    this.encryptionService = encryptionService;
+    this.uuidGenerator = uuidGenerator;
+    this.restClientFactoryFactory = clientFactoryFactory;
+  }
+
+  /**
+   * If the password is not changed, change it. Only call this method if the system is configured to
+   * use basic authentication for Solr. This method will only change the password if it detects that
+   * the default password is still in use. This object is intended to be a singleton, a blueprint
+   * bean. That is why is can use itself as the lock
+   */
+  public synchronized void updateSolrPassword() {
+    if (configuredToAttemptAutoPasswordChange() && isUsingDefaultPassword()) {
+      initialize();
```

Comment:
❓ Should we not clear the boolean flags and responses kept around before starting up and also at the end when done such as not to keep a response object in memory for no reasons?